File: config/raycast/scripts/templates/script-command.template.js
#!/usr/bin/env node

// Raycast Script Command: Javascript (requires Nodejs)
// https://github.com/raycast/script-commands/blob/master/commands
// SOURCE: https://github.com/raycast/script-commands/blob/master/templates/script-command.template.js
// ARGUMENTS: https://github.com/raycast/script-commands/blob/master/documentation/ARGUMENTS.md
// OUTPUTMODES: https://github.com/raycast/script-commands/blob/master/documentation/OUTPUTMODES.md
//
// Required parameters:
// @raycast.schemaVersion 1
// @raycast.title My First Script
// @raycast.mode fullOutput
// @raycast.packageName Raycast Scripts
//
// Optional parameters:
// @raycast.icon 🤖
// @raycast.argument1 { "type": "text", "placeholder": "js, css, html", "optional": true}
// @raycast.argument2 { "type": "text", "placeholder": "query" }
//
// Documentation:
// @raycast.description Write a nice and descriptive summary about your script command here
// @raycast.author roalcantara
// @raycast.authorURL https://github.com/roalcantara
//
/*
	ABOUT THIS TEMPLATE:
	This template demonstrates the following ideas:
	* Extracting passed arguments.
	* Using both required and optional arguments
	* URI Encoding
	* Outputting result to Raycast
	* Opening a url using exec and the unix open command
	* Use of destructuring
	* Use of template literals
*/
const { exec } = require('child_process')

// Use destructuring to grab arguments.
// Use slice to start from position 3.
let [topic, query] = process.argv.slice(2)
let uri = `https://developer.mozilla.org/search?topic=${topic}&q=${encodeURIComponent(query)}`

// console.log() displays output in fullOutput mode.
console.log(`The arguments are: \n   ${process.argv.join('\n   ')}\n`)
console.log(`Your topic is "${topic}"`)
console.log(`Your query is "${query}"`)
console.log(`Your query uri encoded is "${encodeURIComponent(query)}"`)
console.log(`The uri is "${uri}"`)

// Uncomment the exec line below to open this query in your web browser.
// Use double quotes around the uri to avoid processing issues.
//exec(`open "${uri}"`)

// vi: set ft=javascript

File: config/raycast/scripts/templates/script-command.template.py
#!/usr/bin/env python3

# Raycast Script Command: Python (requires Python3)
# https://github.com/raycast/script-commands
# SOURCE: https://github.com/raycast/script-commands/blob/master/templates/script-command.template.php
# ARGUMENTS: https://github.com/raycast/script-commands/blob/master/documentation/ARGUMENTS.md
# OUTPUTMODES: https://github.com/raycast/script-commands/blob/master/documentation/OUTPUTMODES.md
#
# Required parameters:
# @raycast.schemaVersion 1
# @raycast.title String Command Example
# @raycast.mode fullOutput
# @raycast.packageName Raycast Scripts
#
# Optional parameters:
# @raycast.icon 🤖
# @raycast.currentDirectoryPath ~
# @raycast.needsConfirmation false
# @raycast.argument1 {"type": "text", "placeholder": "path/to/file.xml"}
# @raycast.argument2 {"type": "dropdown", "placeholder": "path/to/file.xml", "data": [{"title": "Label1", "value": "value1"}, {"title": "Label2", "value": "value2"}]}
# @raycast.argument3 {"type": "text", "placeholder": "Optional", "optional": true}
# @raycast.argument4 {"type": "password", "placeholder": "Password"}
#
# Documentation:
# @raycast.description Python script command example
# @raycast.author roalcantara
# @raycast.authorURL https://github.com/roalcantara
#
# USAGE:
# python script-command.template.py 'hi' 'there' '/path/to/file.xml' 'password'

import sys
args = sys.argv[1:]  # capture all arguments passed to the script

print("Hello World! " + " ".join(args))

# vi: set ft=python
